Subsurface/Node/Common/Peripherals/Inc/can_manager.h

```diff
@@ -0,0 +1,70 @@
+#ifndef NODE_CAN_MANAGER_H
+#define NODE_CAN_MANAGER_H
+
+#ifndef UNIT_TEST
+#include "can_interface.h"
+#else
+#include "mock_can_interface.h"
+#endif
+
+#include <cstring>
+
+class CANManager
+{
+private:
+    CANInterface* m_can_interface;
+    uint8_t m_tx_buffer[8] = {0};
+    uint8_t m_rx_buffer[8] = {0};
+
+public:
+    CANManager (CANInterface* can_interface);
+
+    bool transmit();
+
+    bool receive();
+
+    void setTxId(uint16_t can_id);
+
+    uint8_t* getTxBuffer() { return m_tx_buffer; }
+    uint8_t* getRxBuffer() { return m_rx_buffer; }
+
+#ifdef UNIT_TEST
+    void setRxBuffer(uint8_t* buffer, int len)
+    {
+        memset(m_rx_buffer, 0, 8);

```

Comment:
technically no, but in the case that the buffer that is passed in has a length of less than 8, then however m_rx_buffer was set before from len to 8 will be whatever it was before. Just for consistency it is nicer I think to clear the buffer here so that we know it contains all zeroes at the start